lib/src/arithmetic/floating_point/floating_point_value.dart

```diff
@@ -0,0 +1,900 @@
+    }
+  }
+
+  /// Convert from double using its native binary representation
+  factory FloatingPointValue.fromDouble(double inDouble,
+      {required int exponentWidth,
+      required int mantissaWidth,
+      FloatingPointRoundingMode roundingMode =
+          FloatingPointRoundingMode.roundNearestEven}) {
+    if ((exponentWidth == 8) && (mantissaWidth == 23)) {
+      return FloatingPoint32Value.fromDouble(inDouble);
+    } else if ((exponentWidth == 11) && (mantissaWidth == 52)) {
+      return FloatingPoint64Value.fromDouble(inDouble);
+    }
+
+    final fp64 = FloatingPoint64Value.fromDouble(inDouble);
+    final exponent64 = fp64.exponent;
+
+    var expVal = (exponent64.toInt() - fp64.bias) +
+        FloatingPointValue.computeBias(exponentWidth);
+    // Handle subnormal
+    final mantissa64 = [
+      if (expVal <= 0)
+        ([LogicValue.one, fp64.mantissa].swizzle() >>> -expVal).slice(52, 1)
+      else
+        fp64.mantissa
+    ].first;
+    var mantissa = mantissa64.slice(51, 51 - mantissaWidth + 1);
+
+    if (roundingMode == FloatingPointRoundingMode.roundNearestEven) {
+      final sticky = mantissa64.slice(51 - (mantissaWidth + 2), 0).or();
+      final roundPos = 51 - (mantissaWidth + 2) + 1;
+      final round = mantissa64[roundPos];
+      final guard = mantissa64[roundPos + 1];
+
+      // RNE Rounding
+      if (guard == LogicValue.one) {
+        if ((round == LogicValue.one) |
+            (sticky == LogicValue.one) |
+            (mantissa[0] == LogicValue.one)) {
+          mantissa += 1;
+          if (mantissa == LogicValue.zero.zeroExtend(mantissa.width)) {
+            expVal += 1;
+          }
+        }
+      }
+    }
+
+    final exponent =
+        LogicValue.ofBigInt(BigInt.from(max(expVal, 0)), exponentWidth);
+
+    return FloatingPointValue(
+        sign: fp64.sign, exponent: exponent, mantissa: mantissa);
+  }
+
+  /// Generate a random [FloatingPointValue], supplying random seed [rv].
+  factory FloatingPointValue.random(Random rv,
+      {required int exponentWidth,
+      required int mantissaWidth,
+      bool normal = false}) {
+    final largestExponent = FloatingPointValue.computeBias(exponentWidth) +
+        FloatingPointValue.computeMaxExponent(exponentWidth);
+    final s = rv.nextLogicValue(width: 1).toInt();
+    var e = BigInt.one;
+    do {
+      e = rv
+          .nextLogicValue(width: exponentWidth, max: largestExponent)
+          .toBigInt();
+    } while ((e == BigInt.zero) & normal);
+    final m = rv.nextLogicValue(width: mantissaWidth).toBigInt();
+    return FloatingPointValue(
+        sign: LogicValue.ofInt(s, 1),
+        exponent: LogicValue.ofBigInt(e, exponentWidth),
+        mantissa: LogicValue.ofBigInt(m, mantissaWidth));
+  }
+
+  /// Convert a floating point number into a [FloatingPointValue]
+  /// representation. This form performs NO ROUNDING.
+  factory FloatingPointValue.fromDoubleIter(double inDouble,
```

Comment:
what is "iter"?

Reply:
This is a converter that iterates in order to convert from double to avoid an overflow.
This routine is general as it does not need to know the format of the native double.
.fromDouble() relies on the exact bit representation of the native double in doing the conversion.

We don't absolutely have to have this routine, but it took a long time to get it to work correctly.

Reply:
It's a good thing to keep, but I'm wondering if there's a better name for this that describes the functionality instead of the implementation?